src/LinAlg/hiopLinSolverSparseGinkgo.hpp

```diff
@@ -0,0 +1,168 @@
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+// OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+// AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+// EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Additional BSD Notice
+// 1. This notice is required to be provided under our contract with the U.S. Department
+// of Energy (DOE). This work was produced at Lawrence Livermore National Laboratory under
+// Contract No. DE-AC52-07NA27344 with the DOE.
+// 2. Neither the United States Government nor Lawrence Livermore National Security, LLC
+// nor any of their employees, makes any warranty, express or implied, or assumes any
+// liability or responsibility for the accuracy, completeness, or usefulness of any
+// information, apparatus, product, or process disclosed, or represents that its use would
+// not infringe privately-owned rights.
+// 3. Also, reference herein to any specific commercial products, process, or services by
+// trade name, trademark, manufacturer or otherwise does not necessarily constitute or
+// imply its endorsement, recommendation, or favoring by the United States Government or
+// Lawrence Livermore National Security, LLC. The views and opinions of authors expressed
+// herein do not necessarily state or reflect those of the United States Government or
+// Lawrence Livermore National Security, LLC, and shall not be used for advertising or
+// product endorsement purposes.
+
+#ifndef HIOP_LINSOLVER_GINKGO
+#define HIOP_LINSOLVER_GINKGO
+
+#include "hiopLinSolver.hpp"
+#include "hiopMatrixSparseTriplet.hpp"
+
+
+#include <ginkgo/ginkgo.hpp>
+
+
+#include <unordered_map>
+
+/** implements the linear solver class using Ginkgo
+ *
+ * @ingroup LinearSolvers
+ */
+namespace hiop {
+
+class hiopLinSolverSymSparseGinkgo: public hiopLinSolverSymSparse
+{
+public:
+  hiopLinSolverSymSparseGinkgo(const int& n, const int& nnz, hiopNlpFormulation* nlp);
+  virtual ~hiopLinSolverSymSparseGinkgo();
+
+  /** Triggers a refactorization of the matrix, if necessary.
+   * Overload from base class. */
+  int matrixChanged();
+
+  /** solves a linear system.
+   * param 'x' is on entry the right hand side(s) of the system to be solved. On
+   * exit is contains the solution(s).  */
+  bool solve ( hiopVector& x_ );
+
+//protected:
+//  int* ipiv;
+//  hiopVector* dwork;
+
+private:
+
+  int      m_;                         // number of rows of the whole matrix
+  int      n_;                         // number of cols of the whole matrix
+  int      nnz_;                       // number of nonzeros in the matrix
+
+  int *index_covert_CSR2Triplet_;
+  int *index_covert_extra_Diag2CSR_;
+
+  int nFakeNegEigs_;
+
```

Comment:
```suggestion
```
"Fake inertia" approach should not be used anymore and this member should be removed.